### PR TITLE
Improve Type extraction for ballerina OpenApi

### PR DESCRIPTION
**Pull request by DimalChandrasiri**

### CODE REVIEW

File: misc/openapi-ballerina/modules/openapi-to-ballerina-generator/src/main/java/org/ballerinalang/openapi/typemodel/BallerinaIOpennApiRequestBody.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package org.ballerinalang.openapi.typemodel;
```
Comment (NipunaMarcus) [inline misc/openapi-ballerina/modules/openapi-to-ballerina-generator/src/main/java/org/ballerinalang/openapi/typemodel/BallerinaIOpennApiRequestBody.java]: class name is BallerinaIOpenApiRequestBody.java? What is "I" between "Ballerina" and "OpenApi" stands for?

Comment (NipunaMarcus) [reply misc/openapi-ballerina/modules/openapi-to-ballerina-generator/src/main/java/org/ballerinalang/openapi/typemodel/BallerinaIOpennApiRequestBody.java]: And there is an additional "n" in the "OpenApi"

Comment (DimalChandrasiri) [reply misc/openapi-ballerina/modules/openapi-to-ballerina-generator/src/main/java/org/ballerinalang/openapi/typemodel/BallerinaIOpennApiRequestBody.java]: FIxed with commit cb03db8803b33e48835ec75daaac998460e16a22

File: misc/openapi-ballerina/modules/openapi-to-ballerina-generator/src/main/java/org/ballerinalang/openapi/typemodel/BallerinaOpenApiComponent.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package org.ballerinalang.openapi.typemodel;
```
Comment (anupama-pathirage) [inline misc/openapi-ballerina/modules/openapi-to-ballerina-generator/src/main/java/org/ballerinalang/openapi/typemodel/BallerinaOpenApiComponent.java]: Missing license header.

File: misc/openapi-ballerina/modules/openapi-to-ballerina-generator/src/main/java/org/ballerinalang/openapi/typemodel/BallerinaOpenApiOperation.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
+package org.ballerinalang.openapi.typemodel;
```
Comment (anupama-pathirage) [inline misc/openapi-ballerina/modules/openapi-to-ballerina-generator/src/main/java/org/ballerinalang/openapi/typemodel/BallerinaOpenApiOperation.java]: Missing license header.

File: misc/openapi-ballerina/modules/openapi-to-ballerina-generator/src/main/java/org/ballerinalang/openapi/typemodel/BallerinaOpenApiParameter.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
+package org.ballerinalang.openapi.typemodel;
```
Comment (anupama-pathirage) [inline misc/openapi-ballerina/modules/openapi-to-ballerina-generator/src/main/java/org/ballerinalang/openapi/typemodel/BallerinaOpenApiParameter.java]: Missing license header. Check all the new classes

File: misc/openapi-ballerina/modules/openapi-to-ballerina-generator/src/test/java/org/ballerinalang/openapi/cmd/OpenApiGenServiceCmdTest.java [hunk 5]
```diff
@@ -275,4 +275,101 @@ public void testInlineRequestBodyServiceGen() throws IOException {
         }
     }
 
+    @Test(description = "Test open-api genservice for successful service generation with all of schema type")
```
Comment (anupama-pathirage) [inline misc/openapi-ballerina/modules/openapi-to-ballerina-generator/src/test/java/org/ballerinalang/openapi/cmd/OpenApiGenServiceCmdTest.java]: It seems we don't have a license header in this file too. Check all the files in the package.

Comment (NipunaMarcus) [reply misc/openapi-ballerina/modules/openapi-to-ballerina-generator/src/test/java/org/ballerinalang/openapi/cmd/OpenApiGenServiceCmdTest.java]: Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/20891

File: misc/openapi-ballerina/modules/openapi-to-ballerina-generator/src/main/java/org/ballerinalang/openapi/typemodel/BallerinaOpenApiParameter.java (hunk 4 above)
Comment (NipunaMarcus) [reply misc/openapi-ballerina/modules/openapi-to-ballerina-generator/src/main/java/org/ballerinalang/openapi/typemodel/BallerinaOpenApiParameter.java]: Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/20891

File: misc/openapi-ballerina/modules/openapi-to-ballerina-generator/src/main/java/org/ballerinalang/openapi/typemodel/BallerinaOpenApiOperation.java (hunk 3 above)
Comment (NipunaMarcus) [reply misc/openapi-ballerina/modules/openapi-to-ballerina-generator/src/main/java/org/ballerinalang/openapi/typemodel/BallerinaOpenApiOperation.java]: Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/20891

File: misc/openapi-ballerina/modules/openapi-to-ballerina-generator/src/main/java/org/ballerinalang/openapi/typemodel/BallerinaOpenApiComponent.java (hunk 2 above)
Comment (NipunaMarcus) [reply misc/openapi-ballerina/modules/openapi-to-ballerina-generator/src/main/java/org/ballerinalang/openapi/typemodel/BallerinaOpenApiComponent.java]: Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/20891

**Pull request merged**